pkg/image/image.go

```diff
@@ -392,26 +393,38 @@ func ChangeImageDomain(imageString string, newDomain string) (types.ImageReferen
 	return newImage, nil
 }
 
-func WithoutTag(image string) (string, error) {
-	fullImage, err := AddDefaultRegistry(image, "place.holder.com")
-	if err != nil {
-		return image, err
+// SplitImage Returns an image tag if it exists, an image digest if it exists, the image without the tag and digest, and any errors; in this order.
+// images with both a tag and digest are not supported
+func SplitImage(imageArg string) (ocneTypes.ImageInfo, error) {
```

Comment:
Moving this comment from a merged PR to this one:

The steps used to split an image is very similar to the steps merged in here:

https://github.com/oracle-cne/ocne/blob/27082345f7310695a124d29d43fd584d2c8d9549/pkg/image/registry.go#L103

The SplitImage function in this PR has unit testing and offers specific error messages.

Should we change either of these code blocks to match each other?

@dkrasins @pfmackin

Reply:
@George-Aeillo From Dan's response [here](https://github.com/oracle-cne/ocne/pull/19#discussion_r1773490703) we should change https://github.com/oracle-cne/ocne/blob/27082345f7310695a124d29d43fd584d2c8d9549/pkg/image/registry.go#L103 to use the new SplitImage function

Reply:
I implemented these changes